src/System/include/BipedalLocomotion/System/ConstantWeightProvider.h

```diff
@@ -41,22 +40,34 @@ struct ConstantWeightProvider : public IWeightProvider
      */
     ConstantWeightProvider(Eigen::Ref<const Eigen::VectorXd> weight);
 
+    /**
+     * Since the weight is constant this will always return true and it does nothing
+     */
+    bool advance() final;
+
     /**
      * Get the weight associated to the provider
      * @return A vector representing the diagonal matrix of the weight
      */
-    Eigen::Ref<const Eigen::VectorXd> getWeight() const final;
+    const Eigen::VectorXd& getOutput() const final;
+
+    /**
+     * Determines the validity of the weight
+     * @return True if the weight is valid, false otherwise.
+     */
+    bool isOutputValid() const final;
 
     /**
      * Initialize constant weight provider.
      * @param handler pointer to the parameter handler.
      * @note The following parameters are required:
-     * |  Parameter Name  |        Type      |                          Description                              | Mandatory |
+     * |  Parameter Name  |        Type      |                          Description | Mandatory |
      * |:----------------:|:----------------:|:-----------------------------------------------------------------:|:---------:|
-     * |    `weight`      | `vector<double>` |  Vector representing the diagonal matrix of a constant weight     |    Yes    |
+     * |    `weight`      | `vector<double>` |  Vector representing the diagonal matrix of a
+     * constant weight     |    Yes    |
```

Comment:
I think that this comment got automatically cut in two lines but it probably shouldn't.